Kontent.Ai.Management/Extensions/AssetExtensions.cs

```diff
@@ -71,6 +71,38 @@ public static AssetFolderHierarchy GetFolderHierarchyByExternalId(this IEnumerab
         }
         return null;
     }
+    
+    /// <summary>
+    /// Gets folder hierarchy for a given folder codename
+    /// </summary>
+    /// <param name="folders">The <see cref="AssetFoldersModel.Folders"/> property retrieved from the <see cref="IManagementClient.GetAssetFoldersAsync"/> method.</param>
+    /// <param name="codename">Folder codename</param>
+    /// <returns>The <see cref="AssetFolderHierarchy"/> instance that represents the folder found for a given folder codename. Null if not found.</returns>
+    public static AssetFolderHierarchy GetFolderHierarchyByCodename(this IEnumerable<AssetFolderHierarchy> folders, string codename)
+    {
+        if (folders == null)
+        {
+            return null;
+        }
+
+        // Recursively search for the folder hierarchy that an asset is in. Returns null if file is not in a folder.
+        foreach (var itm in folders)
+        {
+            if (itm.Codename == codename)
+            {
+                return itm;
+            }
+            else if (itm.Folders != null)
```

Comment:
I would lose this unnecessary `else` statement and use null coalescing to make it a bit more compact. also, wouldn't `folder` be more fitting variable name than `itm` in the foreach? seems slightly more readable, but that's just a nitpick

```csharp
        foreach (var folder in folders)
        {
            if (folder.Id == folderId)
            {
                return folder;
            }

            var nestedFolder = folder.Folders?.GetFolderHierarchyById(folderId);
            if (nestedFolder != null) //This is required so you don't stop processing if the root contains many folders (let the above foreach loop continue)
            {
                return nestedFolder;
            }
        }
```

Reply:
Yeap, looks good, fixed.